src/fsharp/TastOps.fs

```diff
@@ -5433,13 +5433,13 @@ let mapTargetsOfDecisionTree f tree = mapAccTipsOfDecisionTree (fun es i -> TDSu
 
 // Dead target elimination 
 let eliminateDeadTargetsFromMatch tree (targets:_[]) =
-    let used = accTargetsOfDecisionTree tree [] |> ListSet.setify (=) |> Array.ofList
-    if used.Length < targets.Length then
-        Array.sortInPlace used;
+    let used = accTargetsOfDecisionTree tree [] |> HashSet
+    if used.Count < targets.Length then
+        let used = used |> Seq.toArray
```

Comment:
Could we use `Seq.iteri` instead, without converting the set to array?

Reply:
It's used below in array.map again